python/cudf/cudf/core/series.py

```diff
@@ -2140,39 +2140,79 @@ def label_encoding(self, cats, dtype=None, na_sentinel=-1):
         Parameters
         ----------
         values : sequence of input values
-        dtype: numpy.dtype; optional
-               Specifies the output dtype.  If `None` is given, the
-               smallest possible integer dtype (starting with np.int8)
-               is used.
-        na_sentinel : number
+        dtype : numpy.dtype; optional
+            Specifies the output dtype.  If `None` is given, the
+            smallest possible integer dtype (starting with np.int8)
+            is used.
+        na_sentinel : number, default -1
             Value to indicate missing category.
 
         Returns
         -------
         A sequence of encoded labels with value between 0 and n-1 classes(cats)
+
+        Examples
+        --------
+        >>> import cudf
+        >>> s = cudf.Series([1, 2, 3, 4, 10])
+        >>> s.label_encoding([2, 3])
+        0   -1
+        1    0
+        2    1
+        3   -1
+        4   -1
+        dtype: int8
+
+        `na_sentinel` parameter can be used to
+        control the value when there is no encoding.
+
+        >>> s.label_encoding([2, 3], na_sentinel=10)
+        0    10
+        1     0
+        2     1
+        3    10
+        4    10
+        dtype: int8
+
+        When none of `cats` values exist in s, entire
+        Series will be `na_sentinel`.
+
+        >>> s.label_encoding(['a', 'b', 'c'])
+        0   -1
+        1   -1
+        2   -1
+        3   -1
+        4   -1
+        dtype: int8
         """
         from cudf import DataFrame
 
+        def _return_sentinel_series():
+            return Series(
+                utils.scalar_broadcast_to(
+                    na_sentinel, size=len(self), dtype=dtype
+                ),
+                index=self.index,
+                name=None,
+            )
```

Comment:
Wondering if it makes more sense to define a `column_full` function similar to [`np.full`](https://numpy.org/doc/stable/reference/generated/numpy.full.html) and then just use that inside the `label_encoding` function instead of defining this `_return_sentinel_series` function.

There may also be libcudf functions to help with creating the "full" column.

Reply:
Again we can do as a follow up